src/main/java/com/targus/ui/controllers/MapController.java

```diff
@@ -0,0 +1,96 @@
+package com.targus.ui.controllers;
+
+import com.targus.ui.Mediator;
+import com.targus.ui.widgets.PotentialPosition;
+import com.targus.ui.widgets.Sensor;
+import com.targus.ui.widgets.Target;
+import javafx.fxml.FXML;
+import javafx.scene.Group;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.MenuItem;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.Pane;
+import javafx.scene.shape.Circle;
+
+public class MapController {
+    @FXML
+    public Pane mainPane;
+    private Mediator mediator;
+    MenuItem createTarget = new MenuItem("Create Target");
+    MenuItem createPotentialPosition = new MenuItem("Create Potential Position");
+    MenuItem createSensor = new MenuItem("Create Sensor");
+    MenuItem removeSensor = new MenuItem("Remove Sensor");
+
+    @FXML
+    void paneClicked(javafx.scene.input.MouseEvent event) {
+        ContextMenu context = new ContextMenu(createTarget, createPotentialPosition, createSensor, removeSensor);
+
+        if (event.getButton() == MouseButton.SECONDARY) {
+            context.show(mainPane, event.getScreenX(), event.getScreenY());
+            createTarget.setOnAction(actionEvent -> {
+                addChild(new Target(event.getX(), event.getY()));
+                mediator.addTarget(new Target(event.getX(), event.getY()));
+            });
+
+            createPotentialPosition.setOnAction(actionEvent -> {
```

Comment:
This part is still not complete because adding and removing sensors with live fitness value display is still in the process. I will keep your suggestion whilst implementing that feature.